LBH.AdultSocialCare.Api/V1/Controllers/DayCarePackageReclaimController.cs

```diff
@@ -0,0 +1,94 @@
+            _getAllHomeCareReclaimCategoryUseCase = getAllHomeCareReclaimCategoryUseCase;
+            _getAllHomeCareReclaimFromUseCase = getAllHomeCareReclaimFromUseCase;
+        }
+
+        [ProducesResponseType(typeof(DayCarePackageClaimResponse), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status422UnprocessableEntity)]
+        [ProducesDefaultResponseType]
+        [HttpPost]
+        public async Task<ActionResult<DayCarePackageClaimResponse>> CreateDayCarePackageReclaim(
+            DayCarePackageClaimCreationRequest dayCarePackageClaimCreationRequest)
+        {
+            if (dayCarePackageClaimCreationRequest == null)
+            {
+                return BadRequest("Object for creation cannot be null.");
+            }
+
+            if (!ModelState.IsValid)
+            {
+                return UnprocessableEntity(ModelState);
+            }
+
+            var dayCarePackageClaimForCreationDomain = dayCarePackageClaimCreationRequest.ToDomain();
+            var dayCarePackageClaimResponse =
+                await _createDayCarePackageReclaimUseCase.ExecuteAsync(dayCarePackageClaimForCreationDomain).ConfigureAwait(false);
+            return Ok(dayCarePackageClaimResponse);
+        }
+
+        [ProducesResponseType(typeof(IEnumerable<HomeCarePackageReclaimAmountOptionResponse>), StatusCodes.Status200OK)]
+        [ProducesDefaultResponseType]
+        [HttpGet]
+        [Route("amount-option")]
+        public async Task<ActionResult<IEnumerable<HomeCarePackageReclaimAmountOptionResponse>>> GetAmountOptionsList()
+        {
+            var result = await _getAllHomeCareAmountOptionUseCase.GetAllAsync().ConfigureAwait(false);
+            return Ok(result);
+        }
+
+        [ProducesResponseType(typeof(IEnumerable<HomeCarePackageReclaimFromResponse>), StatusCodes.Status200OK)]
+        [ProducesDefaultResponseType]
+        [HttpGet]
+        [Route("reclaim-from")]
+        public async Task<ActionResult<IEnumerable<HomeCarePackageReclaimFromResponse>>> GetTypeOfReclaimFromOptionList()
+        {
+            var result = await _getAllHomeCareReclaimFromUseCase.GetAllAsync().ConfigureAwait(false);
+            return Ok(result);
+        }
+
+        [ProducesResponseType(typeof(IEnumerable<HomeCarePackageReclaimCategoryResponse>), StatusCodes.Status200OK)]
+        [ProducesDefaultResponseType]
+        [HttpGet]
+        [Route("reclaim-category")]
+        public async Task<ActionResult<IEnumerable<HomeCarePackageReclaimCategoryResponse>>> GetTypeOfReclaimCategoryOptionList()
+        {
+            var result = await _getAllHomeCareReclaimCategoryUseCase.GetAllAsync().ConfigureAwait(false);
+            return Ok(result);
+        }
+    }
```

Comment:
I think the naming is intentional. Can we have the common names as `PackageReclaimCategoryResponse`, `PackageReclaimFromResponse`, `PackageReclaimAmountOptionResponse` without a specific care package prefix instead. Same for the domain naming

Reply:
yes definitely, I miss that.